tests/test_basic.py

```diff
@@ -297,3 +297,71 @@ def test_mons_up_down(rook_cluster):
         check += 1
 
     assert mon_pods == mons
+
+
+def test_rbd_pvc(rook_cluster):
+    # create a CephBlockPool
+    output = rook_cluster.kubernetes.kubectl_apply(
+                os.path.join(
+                    rook_cluster.ceph_dir, 'csi/rbd/storageclass.yaml'))
+
+    if output[0] != 0:
+        pytest.fail("Could not create a CephBlockPool StorageClass")
+
+    # check if StorageClass is up and available
+    output = rook_cluster.kubernetes.kubectl('get sc rook-ceph-block')
+
+    if output[0] != 0:
+        pytest.fail("Could not find a CephBlockPool called StorageClass")
+
+    # create an rbd based PVC
+    output = rook_cluster.kubernetes.kubectl_apply(
+                os.path.join(rook_cluster.ceph_dir, 'csi/rbd/pvc.yaml'))
+
+    if output[0] != 0:
+        pytest.fail("Could not create a rbd-PVC")
+
+    pvc = rook_cluster.kubernetes.kubectl('get -o yaml pvc rbd-pvc')
+
+    if pvc[0] != 0:
+        pytest.fail("Could not get rbd-pvc details")
+
+    pvc_object = yaml.load(pvc[1], Loader=yaml.BaseLoader)
```

Comment:
(nit) Alternatively you might be able to use `rook_cluster.kubernetes.v1.list_persistent_volume_claim_for_all_namespaces`

Reply:
I changed the complete logic to use `wait_for_result` instead. This cleans up the code a lot.